src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -2146,42 +2146,56 @@ public boolean getCellSelectionEnabled() {
-     */
-    public void selectAll() {
-        // If I'm currently editing, then I should stop editing
-        if (isEditing()) {
-            removeEditor();
-        }
-        if (getRowCount() > 0 && getColumnCount() > 0) {
-            int oldLead;
-            int oldAnchor;
-            ListSelectionModel selModel;
+    private void selectRows(int rowCount) {
+        ListSelectionModel selModel = selectionModel;
+        selModel.setValueIsAdjusting(true);
+        int oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
+        int oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
 
-            selModel = selectionModel;
-            selModel.setValueIsAdjusting(true);
-            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
-            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
+        setRowSelectionInterval(0, rowCount - 1);
 
-            setRowSelectionInterval(0, getRowCount()-1);
+        // this is done to restore the anchor and lead
+        SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
 
-            // this is done to restore the anchor and lead
-            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+        selModel.setValueIsAdjusting(false);
+    }
 
-            selModel.setValueIsAdjusting(false);
+    private void selectColumns(int columnCount) {
+        ListSelectionModel selModel = columnModel.getSelectionModel();
+        selModel.setValueIsAdjusting(true);
+        int oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), false);
+        int oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), false);
 
-            selModel = columnModel.getSelectionModel();
-            selModel.setValueIsAdjusting(true);
-            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), false);
-            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), false);
+        setColumnSelectionInterval(0, columnCount - 1);
 
-            setColumnSelectionInterval(0, getColumnCount()-1);
+        // this is done to restore the anchor and lead
+        SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+
+        selModel.setValueIsAdjusting(false);
+    }
 
-            // this is done to restore the anchor and lead
-            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+    /**
+     *  Selects all rows, columns, and cells in the table.
+     */
+    public void selectAll() {
+        int rowCount = getRowCount();
+        int columnCount = getColumnCount();
 
-            selModel.setValueIsAdjusting(false);
+        // If I'm currently editing, then I should stop editing
+        if (isEditing()) {
+            removeEditor();
+        }
+        if (rowCount > 0 && columnCount > 0) {
+            selectRows(rowCount);
+            selectColumns(columnCount);
```

Comment:
Here the checks are missing. Both row and column selection allowed check.

Reply:
I know but it is not there in the code before the fix too..THere's no bug raised for that in JBS so I would like to continue with that for now so that we dont get any difference in behavior..I added the check in the new code..